org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/binding/BindingsView.java

```diff
@@ -252,7 +262,51 @@ public void connectionRemoved(IConnectedEngineFacade server) {
     for (var server : SonarLintCorePlugin.getServersManager().getServers()) {
       server.addConnectedEngineListener(serverListener);
     }
-
+    
+    // add listener for when project is opened / closed / deleted
+    projectListener = event -> {
+      if (event.getType() == IResourceChangeEvent.PRE_CLOSE
```

Comment:
Why not declaring the `shouldRefresh` variable here, instead of risking a race condition?

Reply:
Cannot be done 😞 When used as well in `IResourceDelta.accept(...)` in line 273+ it cannot be compiled with the issue: "Variable shouldRefresh used in lambda expression should be final or effectively final".
The ideal resolution would be to use a class variable and in case of the listener being delegated by the workspace to different threads (which I cannot foresee) to use `volatile`.

@henryju regarding a race condition: From manual testing, it seems that the events are processed via a queue but I'm not sure if it is really the case. I can move it to an `AtomicBoolean` maybe?

Reply:
Yeah, I think a local AtomicBoolean is better in my opinion

Reply:
Done